pallets/utxo/src/tests.rs

```diff
@@ -740,3 +681,814 @@ proptest! {
+    // Ticker empty
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: vec![],
+        amount_to_issue: 0,
+        number_of_decimals: 0,
+        metadata_uri: vec![],
+    };
+    test_tx!(data, Err, "token ticker can't be empty");
+}
+
+#[test]
+fn test_tokens_issuance_too_big_ticker() {
+    // Ticker too long
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: Vec::from([0u8; 10_000]),
+        amount_to_issue: 0,
+        number_of_decimals: 0,
+        metadata_uri: vec![],
+    };
+    test_tx!(data, Err, "token ticker is too long");
+}
+
+#[test]
+fn test_tokens_issuance_amount_zero() {
+    // Amount to issue is zero
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: b"BensT".to_vec(),
+        amount_to_issue: 0,
+        number_of_decimals: 0,
+        metadata_uri: vec![],
+    };
+    test_tx!(data, Err, "output value must be nonzero");
+}
+
+#[test]
+fn test_tokens_issuance_too_big_decimals() {
+    // Number of decimals more than 18 numbers
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: b"BensT".to_vec(),
+        amount_to_issue: 1_000_000_000,
+        number_of_decimals: 19,
+        metadata_uri: vec![],
+    };
+    test_tx!(data, Err, "too long decimals");
+}
+
+#[test]
+fn test_tokens_issuance_empty_metadata() {
+    // metadata_uri empty
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: b"BensT".to_vec(),
+        amount_to_issue: 1_000_000_000,
+        number_of_decimals: 18,
+        metadata_uri: vec![],
+    };
+    test_tx!(data, Ok, "");
+}
+
+#[test]
+fn test_tokens_issuance_too_long_metadata() {
+    // metadata_uri too long
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: b"BensT".to_vec(),
+        amount_to_issue: 1_000_000_000,
+        number_of_decimals: 18,
+        metadata_uri: Vec::from([0u8; 10_000]),
+    };
+    test_tx!(data, Err, "token metadata uri is too long");
+}
+
+#[test]
+fn test_tokens_issuance_with_junk_data() {
+    // The data field of the maximum allowed length filled with random garbage
+    let mut rng = rand::thread_rng();
+    let garbage = build_random_vec(100);
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: vec![0, 255, 254, 2, 1],
+        amount_to_issue: rng.gen::<u64>() as u128,
+        number_of_decimals: 18,
+        metadata_uri: garbage.clone(),
+    };
+    test_tx!(data, Err, "token ticker has none ascii characters");
+}
+
+#[test]
+fn test_tokens_issuance_with_corrupted_uri() {
+    let mut rng = rand::thread_rng();
+    let garbage = build_random_vec(100);
+    // garbage uri
+    let data = OutputData::TokenIssuanceV1 {
+        token_ticker: b"BensT".to_vec(),
+        amount_to_issue: rng.gen::<u64>() as u128,
+        number_of_decimals: 18,
+        metadata_uri: garbage,
+    };
+    test_tx!(data, Err, "metadata uri has none ascii characters");
+}
+
+#[test]
+fn test_two_token_creation_in_one_tx() {
```

Comment:
Please also test transfer + issuance in one transaction. It's much easier to limit all transactions to have only issueance OR transfers/burns. We can study later whether we can remove this limitation.

Reply:
Added a new test: `test_transfer_and_issuance_in_one_tx`. Done!